src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -1000,6 +1000,11 @@ Copyright (c) .NET Foundation. All rights reserved.
     <ProjectCapability Remove="ReferenceManagerAssemblies" />
   </ItemGroup>
 
+  <!-- Enable hot reload in 6.0 and newer apps by default -->
+  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' AND $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), '6.0'))">
+    <ProjectCapability Include="SupportsHotReload" />
+  </ItemGroup>
```

Comment:
@dsplaisted since this change touches the NET.SDK.


@BillHiebert this change adds a new project capability dotnet-watch would use to determine if a project can do hot reload.  My hope was that we could also use this in VS rather than rely on TFMs. How does that sound?

Reply:
The implication here is that we expect hot reload to work by default on any and every app targeting .NETCoreApp 6.0 and above, correct?

Reply:
@pranavkm using a capability rather than checking the tfm is fine by me.

Reply:
/cc @tommcdon / @DamianEdwards in case you think we should narrow this down. But we generally think all .NET 6 projects would be eligible to be hot reloaded.

Reply:
For example, will hot reload work on a Xamarin app running on an Android simulator so long as it targets 6.0?

Reply:
The current plan is for that to work (and for dotnet-watch to support this) by of 6.0. We could narrow it down to the project types that are currently supported (WinForms, WPF, Web, and Blazor) if that's not true closer to 6.0.